### PR TITLE
display progress during benchmark

### PR DESCRIPTION
**Pull request by song-zhang**
$ ./bbolt bench -batch-size 10 -count 10000 -key-size 100 -path bolt.db -value-size 200 -write-mode rnd
starting write benchmark.
Completed 350 requests, 349.9/s 
Completed 760 requests, 409.5/s 
Completed 1150 requests, 389.1/s 
Completed 1560 requests, 411.3/s 
Completed 1970 requests, 410.1/s 
Completed 2420 requests, 449.3/s 
Completed 2870 requests, 450.8/s 
Completed 3290 requests, 419.9/s 
Completed 3730 requests, 440.1/s 
Completed 4150 requests, 418.5/s 
Completed 4590 requests, 441.0/s 
Completed 5030 requests, 440.6/s

### CODE REVIEW

File: cmd/bbolt/main.go [hunk 1]
```diff
@@ -1312,19 +1312,22 @@ func (cmd *BenchCommand) Run(args ...string) error {
 	defer db.Close()
 
 	// Write to the database.
-	var results BenchResults
-	if err := cmd.runWrites(db, options, &results); err != nil {
+	writeResults := BenchResults{int64(0), 0}
+	fmt.Fprintf(cmd.Stdout, "starting write benchmark.\n")
```
Comment (ptabor) [inline cmd/bbolt/main.go:1316]: Why not to stick to a single output stream (i.e. Stderr) ?

Comment (song-zhang) [reply cmd/bbolt/main.go]: update, use Stderr as single output stream

File: cmd/bbolt/main.go [hunk 2]
```diff
@@ -1604,6 +1611,23 @@ func (cmd *BenchCommand) runReadsSequentialNested(db *bolt.DB, options *BenchOpt
...
+	ticker := time.Tick(time.Second)
+	lastCompleted, lastTime := int64(0), time.Now()
+	for {
+		select {
+		case <-finishChan:
+			return
+		case t := <-ticker:
+			completed, taken := results.CompletedOps, t.Sub(lastTime)
+			fmt.Fprintf(cmd.Stderr, "Completed %d requests, %3.1f/s \n",
+				completed, float64(completed-lastCompleted)*float64(int64(time.Second))/float64(int64(taken)),
```
Comment (ptabor) [inline cmd/bbolt/main.go:1624]: does the double cost (int -> float) helps here ?

**Review by ptabor — APPROVED** [review]

File: cmd/bbolt/main.go (hunk 2 above)
Comment (song-zhang) [reply cmd/bbolt/main.go]: removed unnecessary conversion. but 'completed'  is int64 type, int support only 2 billions writes.

File: cmd/bbolt/main.go [hunk 3]
```diff
@@ -1431,14 +1438,13 @@ func (cmd *BenchCommand) runWritesRandomNested(db *bolt.DB, options *BenchOption
 }
 
 func (cmd *BenchCommand) runWritesWithSource(db *bolt.DB, options *BenchOptions, results *BenchResults, keySource func() uint32) error {
-	results.WriteOps = options.Iterations
 
-	for i := 0; i < options.Iterations; i += options.BatchSize {
+	for i := int64(0); i < options.Iterations; i += options.BatchSize {
```
Comment (lilic) [inline cmd/bbolt/main.go:1442]: Could we add a comment here on why the need for the conversion, I believe you mentioned due to int limitations, correct?

**Review by lilic — APPROVED** [review]
Thank you, one suggestion.

**Comment by ahrtr** [comment]
I don't understand why this PR isn't merged since there are two approvals. 

Anyway, please rebase this PR since we added some github workflow recently. @song-zhang

**Comment by ahrtr** [comment]
Please
1. squash the commits into one. At least, the second commit https://github.com/etcd-io/bbolt/pull/284/commits/c4cf23dacc10967825aa42aebf6623ae857464b8 doesn't make any sense. 
2. Rebase this PR.

Added label `resurrection` because there is no response  from the original author. Please anyone else feel free to resurrect this PR.

**Comment by cenkalti** [comment]
Resurrected in #420

**Comment by ahrtr** [comment]
superseded by https://github.com/etcd-io/bbolt/pull/420

**Pull request closed without merging**